pkg/netceptor/conn_internal_test.go

```diff
@@ -0,0 +1,870 @@
+		}
+	})
+
+	// Test case 2: Context cancelled
+	t.Run("Context cancelled", func(t *testing.T) {
+		acceptChan := make(chan *AcceptResult)
+		doneChan := make(chan struct{})
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel() // Cancel immediately
+
+		li := &Listener{
+			AcceptChan: acceptChan,
+			DoneChan:   doneChan,
+		}
+
+		// Call the method being tested
+		li.sendResult(ctx, nil, nil)
+
+		// Check that nothing was sent to the channel
+		// We use a non-blocking read to check if anything was sent
+		select {
+		case <-acceptChan:
+			t.Error("Expected no result to be sent due to cancelled context")
+		default:
+			// This is the expected path - nothing was sent
+		}
+	})
+
+	// Test case 3: Listener closed
+	t.Run("Listener closed", func(t *testing.T) {
+		acceptChan := make(chan *AcceptResult)
+		doneChan := make(chan struct{})
+		close(doneChan) // Close the done channel to simulate listener closed
+		ctx := context.Background()
+
+		li := &Listener{
+			AcceptChan: acceptChan,
+			DoneChan:   doneChan,
+		}
+
+		// Call the method being tested
+		li.sendResult(ctx, nil, nil)
+
+		// Check that nothing was sent to the channel
+		// We use a non-blocking read to check if anything was sent
+		select {
+		case <-acceptChan:
+			t.Error("Expected no result to be sent due to closed listener")
+		default:
+			// This is the expected path - nothing was sent
+		}
+	})
+
+	// Test case 4: Successful connection with nil error
+	t.Run("Successful connection", func(t *testing.T) {
+		acceptChan := make(chan *AcceptResult, 1)
+		doneChan := make(chan struct{})
+		ctx := context.Background()
+
+		li := &Listener{
+			AcceptChan: acceptChan,
+			DoneChan:   doneChan,
+		}
+
+		testConn := &mockConn{}
+
+		// Call the method being tested
+		li.sendResult(ctx, testConn, nil)
+
+		// Check if result was sent to the channel
+		select {
+		case result := <-acceptChan:
+			// Verify the result content
+			if result.Conn != testConn {
+				t.Errorf("Expected conn %v, got %v", testConn, result.Conn)
+			}
+			if result.Err != nil {
+				t.Errorf("Expected nil error, got %v", result.Err)
+			}
+		case <-time.After(100 * time.Millisecond):
+			t.Error("Expected result to be sent, but it wasn't")
+		}
+	})
+}
+
+// mockNetceptorForListen is a mock for Netceptor used in listen tests.
+type mockNetceptorForListen struct {
+	listenerRegistry  map[string]*PacketConn
+	reservedServices  map[string]func(*MessageData) error
+	nodeID            string
+	maxForwardingHops byte
+	logger            *logger.ReceptorLogger
+	context           context.Context
+	cancelFunc        context.CancelFunc
+	listenerLock      sync.RWMutex
+	getEphemeralFunc  func() string
+}
+
+// listen is a simplified implementation of Netceptor.listen for testing.
+func (m *mockNetceptorForListen) listen(ctx context.Context, service string, tlscfg *tls.Config, advertise bool, adTags map[string]string) (*Listener, error) {
```

Comment:
But if we mock the `listen` method, we won't be running the real one from `TestListen`, right?